openeo_pg_parser_networkx/pg_schema.py

```diff
@@ -144,63 +147,141 @@ def polygon(self) -> Polygon:
         )
 
 
-class Year(
-    BaseModel
-):  # a more general option would be: [0-9]{4}, but I assume we want years from 1900 to 2100?
-    __root__: str = Field(regex=r"(19|20)[0-9]{2}", max_length=4)
+class Date(BaseModel):
+    __root__: datetime.datetime
 
+    @validator("__root__", pre=True)
+    def validate_time(cls, value: Any) -> Any:
+        if (
+            isinstance(value, str)
+            and len(value) <= 11
+            and match(r"[0-9]{4}[-/][0-9]{2}[-/][0-9]{2}T?", value)
+        ):
+            return pendulum.parse(value)
+        raise ValidationError("Could not parse `Date` from input.")
 
-class Date(BaseModel):
-    __root__: str = Field(regex=r"[0-9]{4}[-/][0-9]{2}[-/][0-9]{2}T?", max_length=11)
+    def to_numpy(self):
+        raise NotImplementedError
+
+    def __repr__(self):
+        return self.__root__.__repr__()
 
 
 class DateTime(BaseModel):
-    __root__: str = Field(
-        regex=r"[0-9]{4}-[0-9]{2}-[0-9]{2}T?[0-9]{2}:[0-9]{2}:?([0-9]{2})?Z?",
-        min_length=15,
-        max_length=20,
-    )
+    __root__: datetime.datetime
 
+    @validator("__root__", pre=True)
+    def validate_time(cls, value: Any) -> Any:
+        if isinstance(value, str) and match(
+            r"[0-9]{4}-[0-9]{2}-[0-9]{2}T?[0-9]{2}:[0-9]{2}:?([0-9]{2})?Z?", value
+        ):
+            return pendulum.parse(value)
+        raise ValidationError("Could not parse `DateTime` from input.")
 
-class Duration(BaseModel):
-    __root__: str = Field(regex=r"P[0-9]*Y?[0-9]*M?[0-9]*D?T?[0-9]*H?[0-9]*M?[0-9]*S?")
+    def to_numpy(self):
+        return np.datetime64(self.__root__)
 
+    def __repr__(self):
+        return self.__root__.__repr__()
 
-class Features(BaseModel):
-    id: Optional[str]
-    type: str
-    geometry: dict
-    properties: Optional[dict]
 
+class Time(BaseModel):
+    __root__: pendulum.Time
 
-GeoJson = Union[FeatureCollection, Feature, GeometryCollection, MultiPolygon, Polygon]
-# The GeoJson spec (https://www.rfc-editor.org/rfc/rfc7946.html#ref-GJ2008) doesn't
-# have a crs field anymore and recommends assuming it to be EPSG:4326, so we do the same.
+    @validator("__root__", pre=True)
+    def validate_time(cls, value: Any) -> Any:
+        if (
+            isinstance(value, str)
+            and len(value) >= 5
+            and len(value) <= 9
+            and match(r"[0-9]{2}:[0-9]{2}:?([0-9]{2})?Z?", value)
+        ):
+            return pendulum.parse(value).time()
+        raise ValidationError("Could not parse `Time` from input.")
 
+    def to_numpy(self):
+        raise NotImplementedError
 
-class JobId(BaseModel):
```

Comment:
Do we want to delete the job-id here? It can also simply be treated like a string, but is there a specific reason to delete it?

Reply:
Ah, not actually deleted, just changed the order of the definitions such that all the temporal ones live together ;)
jobid now lives on line 276!

Reply:
Oh sorry, I did not see it there :) The changes look fine to me!